### PR TITLE
Add Wrap for auto-tuned Rate Limiter

### PR DESCRIPTION
**Pull request by gripleaf**

### CODE REVIEW

File: librocksdb_sys/crocksdb/crocksdb/c.h [hunk 1]
```diff
@@ -1099,6 +1099,14 @@ extern C_ROCKSDB_LIBRARY_API void crocksdb_options_set_delayed_write_rate(
 /* RateLimiter */
 extern C_ROCKSDB_LIBRARY_API crocksdb_ratelimiter_t* crocksdb_ratelimiter_create(
     int64_t rate_bytes_per_sec, int64_t refill_period_us, int32_t fairness);
+typedef enum crocksdb_ratelimiter_mode_t {
```
Comment (huachaohuang) [inline librocksdb_sys/crocksdb/crocksdb/c.h]: We tend to define types at the beginning of this header file, can you move it there?

**Comment by huachaohuang** [comment]
BTW, you need to format the code, try to run `make format`.

**Comment by gripleaf** [comment]
> BTW, you need to format the code, try to run `make format`.

It seems that rust-fmt is supported in rust 1.31.0 & OSX, and I use rust-fmt in stable version instead.

**Comment by DorianZheng** [comment]
Thanks @gripleaf for the PR, I think `rust-toolchain` in the root of the project can specify the toolchain version for you

**Pull request merged**